### PR TITLE
feat: add skeletons on squad directories

### PR DESCRIPTION
**Pull request by ilasw**
## Changes
- Add skeletons on squad directories

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
Discover:  https://mi-540-loading-skeletons-in-squa.preview.app.daily.dev/squads/discover
Featured:  https://mi-540-loading-skeletons-in-squa.preview.app.daily.dev/squads/discover/featured
Caregory: https://mi-540-loading-skeletons-in-squa.preview.app.daily.dev/squads/discover/languages

### CODE REVIEW

File: packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx [hunk 1]
```diff
@@ -68,6 +70,22 @@ export function SquadsDirectoryFeed({
...
+            isFeatured={query.featured}
+          />
+          <PlaceholderSquadGrid
+            className="max-w-80"
+            isFeatured={query.featured}
+          />
+          <PlaceholderSquadGrid
+            className="max-w-80"
+            isFeatured={query.featured}
+          />
```
Comment (nensidosari) [inline packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx:86]: The placeholder should not be based only on query.featured on squad directory feed, related to this issue https://dailydotdev.slack.com/archives/C069XNYMDGV/p1726757281431199

**Review by nensidosari — COMMENTED** [review]
Looks good, 2 minor things

File: packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx (hunk 1 above)
Comment (ilasw) [reply packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx]: This is not the single `featured` coming from the squad's object. 
This comes from the query settings, so if `query.featured` is true there will be only featured cards 👀

Comment (nensidosari) [reply packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx]: Ah cool then just making sure

**Review by nensidosari — APPROVED** [review]

**Comment by ilasw** [comment]
@nensidosari currently there are 6 placeholders them, updated even in this page since using one component to render 6x at once ✅

File: packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx [hunk 2]
```diff
@@ -68,6 +70,12 @@ export function SquadsDirectoryFeed({
           />
         ),
       )}
+      {isInitialLoading && !flatSources.length && (
+        <PlaceholderSquadGridList
+          className="max-w-80"
+          isFeatured={query.featured}
+        />
+      )}
```
Comment (sshanzel) [inline packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx:78]: I think we may want this on mobile too.

**Review by sshanzel — COMMENTED** [review]
Just need to have this on mobile too, but everything else looked great.

File: packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}: PlaceholderSquadGridProps): ReactElement => {
+  const descriptionLength = isFeatured ? 5 : 2;
+  const textLines = Array.from({ length: descriptionLength }).map((_, i) => i);
+
+  return (
+    <div
+      {...attrs}
+      aria-busy
+      className={classNames(
+        'w-[19rem] overflow-hidden rounded-16 border-0 bg-background-subtle p-4',
```
Comment (nensidosari) [inline packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx:33]:
```suggestion
        'overflow-hidden rounded-16 border-0 bg-background-subtle p-4',
```

File: packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}: PlaceholderSquadGridProps): ReactElement => {
+  const descriptionLength = isFeatured ? 5 : 2;
+  const textLines = Array.from({ length: descriptionLength }).map((_, i) => i);
+
+  return (
+    <div
+      {...attrs}
+      aria-busy
+      className={classNames(
+        'overflow-hidden rounded-16 border-0 bg-background-subtle p-4',
```
Comment (nensidosari) [inline packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx:33]:
```suggestion
        'w-80 overflow-hidden rounded-16 border-0 bg-background-subtle p-4',
```

File: packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+}: PlaceholderSquadGridProps): ReactElement => {
+  const descriptionLength = isFeatured ? 5 : 2;
+  const textLines = Array.from({ length: descriptionLength }).map((_, i) => i);
+
+  return (
+    <div
+      {...attrs}
+      aria-busy
+      className={classNames(
+        'w-80 overflow-hidden rounded-16 border-0 bg-background-subtle p-4',
```
Comment (nensidosari) [inline packages/shared/src/components/cards/squad/PlaceholderSquadGrid.tsx:33]: I tried to fix this through my phone but needs the laptop, this implementation does not count for few fixes that were done within the categories itself in my last pr. The card itself is not anymore a fixed width. It is given a fixed width on the discover page, while in others it takes the width from the feed container.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Not sure where to place it:
Generic we have a layout shift due to sidebar: 
https://dailydotdev.slack.com/archives/C069XNYMDGV/p1726811874644229

On the features/category one the placeholders are not actually in the spot the cards end up rendering, this is something we need to fix.
[move-cards.webm](https://github.com/user-attachments/assets/84f8fa6d-ef53-4040-9cc0-6e66ba62f4a8)

**Comment by nensidosari** [comment]
> On the features/category one the placeholders are not actually in the spot the cards end up rendering, this is something we need to fix.
[move-cards.webm](https://github.com/user-attachments/assets/84f8fa6d-ef53-4040-9cc0-6e66ba62f4a8)

@rebelchris this was from my PR merging yesterday, as the cards in the categories now are within the feed container wrapper, while the ones actually being rendered are. 

Have mentioned it here, am taking a look now to fix it https://github.com/dailydotdev/apps/pull/3560#discussion_r1767569893

File: packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx (hunk 2 above)
Comment (ilasw) [reply packages/shared/src/components/cards/squad/SquadsDirectoryFeed.tsx]: Added! ✅

**Review by nensidosari — CHANGES_REQUESTED** [review]
the mobile placeholder is also showing on desktop before the desktop one shows

**Review by nensidosari — APPROVED** [review]
Looks good now!!

**Review by rebelchris — APPROVED** [review]

**Pull request merged**